controllers/space/space_controller.go

```diff
@@ -340,7 +340,10 @@ func (r *Reconciler) manageNSTemplateSet(logger logr.Logger, space *toolchainv1a
 			}
 			nsTmplSet = NewNSTemplateSet(memberCluster.OperatorNamespace, space, spaceBindings.Items, tmplTier)
 			if err := memberCluster.Client.Create(context.TODO(), nsTmplSet); err != nil {
-				logger.Error(err, "failed to create NSTemplateSet on target member cluster")
+				if errors.IsAlreadyExists(err) {
+					// move to next part but don't incremenent space count again (assuming it was already incremented before)
+					goto nstmpl_exists
```

Comment:
or if the usage of `goto` is not a good idea, we could do something like this:
```
return nsTmplSet, norequeue, nil
```
to immediately trigger another reconcile loop without changing the current status (ie, no call to `r.setStatusProvisioning(space)` since it should already be the in this status).

Reply:
yeah it doesn't seem a good practice to use `goto` . I remember it was a bad practice since I was learning COBOL 😄 . 

But I think 1 single `goto` is not bad, better then reconciling IMO! 👍

Reply:
yes indeed, we don't use `goto` anywhere else in our codebase, which is why I'm a bit reluctant, but I thought it could solve the problem without too much refactoring. As I said, the alternative would be to return and trigger a new reconcile loop, as we do in other cases

Reply:
Checking what happens after `nstmpl_exists` I think it will still be reconciled since the `ConditionReady` will not be present in the current object right?

```
_, found := condition.FindConditionByType(nsTmplSet.Status.Conditions, toolchainv1alpha1.ConditionReady)
// skip until there's a `Ready` condition
if !found {
     // just created, but there is no `Ready` condition yet
     return nsTmplSet, requeueDelay, nil
}
```

Reply:
ah right, good point! So I think it would be safer to use `return nsTmplSet, norequeue, nil` instead of `goto` and make sure we ignore the creation in the subsequent loops and move on properly.

Reply:
refactored in 5aecb03